src/targets/gpu/fuse_mlir.cpp

```diff
@@ -190,6 +191,57 @@ struct find_mlir_op
         return {new_gemm_based_op, top_inputs};
     }
 
+    std::string create_mlir_module_name(const module* pointwise_mod,
+                                        instruction_ref gemm_based_op) const
+    {
+        std::string name = "mlir_" + gemm_based_op->name();
+        if(gemm_based_op->name() == "dot")
+        {
+            auto a_shape = gemm_based_op->inputs().front()->get_shape();
+            auto b_shape = gemm_based_op->inputs().back()->get_shape();
+
+            auto m = a_shape.lens()[a_shape.lens().size() - 2];
+            name += "_m" + std::to_string(m);
+            auto k = a_shape.lens().back();
+            name += "_k" + std::to_string(k);
+            auto n = b_shape.lens().back();
+            name += "_n" + std::to_string(n);
```

Comment:
Dont add the sizes to the MLIR symbol name as this makes the summary list each dot or convolution seperately, which the point of the summary is to show the times for the types of operators and not for each instruction, which is what the perf report already prints out.

Reply:
> I was more after getting symbol names in rocprof trace more detailed rather than the perf summary, 

The perf report shows the times for each instruction and you can see the sizes. Also, if you want to use rocprof, we do have `driver roctx` which will insert markers with information about which instruction was executed. Although, I think the perf report is much easier to use.

>  Maybe I can hide this under an option ?

Yea you can add an environment variable.

> From a kernel generator PoV, it would be useful to see breakdown of gemm op rather than collating unrelated gemms

Yea thats why we show the times for each instruction.

Reply:
>  we do have driver roctx
 
 I wasn't aware of this. I ll have a go and see if it suffices.